### PR TITLE
[PS-1872] remember me label triggers toggle

### PR DESCRIPTION
**Pull request by jcuenca95**
## Type of change

```
- [x] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Response to issue #3945 "Remember email" checkbox label isn't clickable 
Clicking on the label of "Remember email" checkbox now toggles the checkbox

## Code changes

Added a member function to toggle the checkbox and call it on label click
This functions just call setValue on the rememberMe formControl  with negated value.

- **file.ext:** Description of what was changed and why

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

**Comment by fromaline** [comment]
It’s a workaround rather than a proper fix. Check the accessibility guidelines, checkbox’s labels should have a proper `for` attribute.

**Review by djsmith85 — CHANGES_REQUESTED** [review]
@jcuenca95 Thank you for your contribution. As mentioned by @fromaline this is currently not the preferred way of solving this.

Since you opened your PR, there has been an addition to our Component library to include a [checkbox component](https://components.bitwarden.com/?path=/docs/component-library-form-checkbox--default). We would like to see that used, which will also fix the linked issue.

Please let us know, if you'd like to follow up on the requested changes.

**Comment by jcuenca95** [comment]
Hi @djsmith85 thanks for the comments. I've updated the changes using the bitCheckbox as you say. I'm happy to help and open to new comments.

File: apps/web/src/app/accounts/login/login.component.html [hunk 1]
```diff
@@ -32,18 +32,16 @@
...
-                bitInput
+                bitCheckbox
                 type="checkbox"
                 formControlName="rememberEmail"
               />
-            </div>
-            <bit-label class="ml-2">
               {{ "rememberEmail" | i18n }}
-            </bit-label>
+            </label>
```
Comment (coroiu) [inline apps/web/src/app/accounts/login/login.component.html:44]:
```
<bit-form-control>
  <input
    id="login_input_remember-email"
    bitCheckbox
    type="checkbox"
    formControlName="rememberEmail"
  />
  <bit-label class="ml-2">
    {{ "rememberEmail" | i18n }}
  </bit-label>
</bit-form-control>```

**Comment by coroiu** [comment]
`CheckboxModule` needs to be imported in the `LoginModule` for the directive to be applied. However I've noticed an issue where the directive styles don't get applied properly due to CSP. I'm looking into it

**Comment by coroiu** [comment]
Fix can be found here: https://github.com/bitwarden/clients/pull/4219 . After that PR gets merged `bitCheckbox` should work as expected! The directive should contain all visual css rules, so we should be able to remove `class="tw-w-4 tw-rounded tw-border"` as well.

Also, if possible, please consider using `bit-form-control` as it is used in the `default` story linked by Daniel above. 

```
  <bit-form-control>
    <input type="checkbox" bitCheckbox formControlName="checkbox" />
    <bit-label>Click me</bit-label>
  </bit-form-control>
```

note: This might add some margin below the checkbox, feel free to adjust it on the `bit-form-control`.

**Review by djsmith85 — CHANGES_REQUESTED** [review]
@jcuenca95 The PR mentioned by @coroiu has been merged. Please have a look at his feedback and looking forward to seeing your changes.

**Review by djsmith85 — CHANGES_REQUESTED** [review]
@jcuenca95 Thank you for updating this PR. Please address the feedback that @coroiu has mentioned with [comment](https://github.com/bitwarden/clients/pull/4046#issuecomment-1347871488) and [comment](https://github.com/bitwarden/clients/pull/4046#issuecomment-1348763221) (I have added a suggestion for the margin fix)

After the feedback has been applied, I think this is ready for the last review and can get passed on to QA.

**Pull request merged**